website/docs/reference/administrative-commands/createIndexes.mdx

```diff
@@ -0,0 +1,134 @@
+---
+sidebar_position: 1
+description: Create one or more indexes on a collection to improve query performance and enable certain query features.
+---
+
+# `createIndexes`
+
+import CodeBlock from '@theme/CodeBlock';
+import CreateIndex from '!!raw-loader!../../guides/create-indexes/1-create-simple-index.request.js';
+import CreateUniqueIndex from '!!raw-loader!../../guides/create-indexes/2-create-unique-index.request.js';
+import CreateCompoundIndex from '!!raw-loader!../../guides/create-indexes/3-create-compound-index.request.js';
+import CreatePartialIndex from '!!raw-loader!../../guides/create-indexes/4-create-partial-index.request.js';
+
+<!-- Category for Command "Administrative, Diagnostic, and Utility Commands" should be organized per folder as present in the compatibility document -->
+
+<!-- Short description of the command and its usage. -->
+
+Create one or more indexes on a given collection to improve query performance or enable certain query features (e.g., full-text, vector, and TTL indexes).
+
+<!-- Syntax structure of the command -->
+
+<!-- markdownlint-disable MD033 -->
+
+This command accepts a collection name and an array of index specification documents.
+The index specification document specifies the fields to be indexed, the index name, and other options such as uniqueness, partial filter expressions, and expiration times for TTL indexes.
+
+```js
+db.runCommand({
+  createIndexes: '<collection_name>',
+  indexes: [
+    {
+      key: {
+        '<field1>': 1, // ascending
+        '<field2>': -1, // descending
+        '<field3>': 'text' // for full-text search
+        // Add more fields as needed
+      },
+      name: '<index_name>', // string
+      unique: false, // Optional
+      partialFilterExpression: {
+        '<field>': { $gte: '<value>' }
+      }, // Optional
+      sparse: false // Optional
+    },
+    // Text index for full-text search
+    {
+      key: { '<field>': 'text' },
+      name: 'text_index_name'
+    },
+
+    // TTL index for automatic expiry
+    {
+      key: { '<timestamp_field>': 1 },
+      name: '<ttl_index_name>',
+      expireAfterSeconds: 3600 // time to expiry in seconds
+    },
+    // Vector index
+    {
+      key: { '<vector_field>': 'cosmosSearch' },
+      name: '<vector_index_name>',
+      cosmosSearchOptions: {
+        kind: 'vector-hnsw',
+        dimensions: 3,
+        similarity: 'COS'
+      }
+    }
+    // Add more index specifications here
+  ]
+})
+```
+
+<!-- majorly fields such as arguments, modifers, stage, operators, parameters, options supported by the command in a table (if necessary or available). Takes in the following parameters: -->
+
+| Field                     | Type     | Description                                                                                                                                                                                                           |
```

Comment:
Can we have a column for specifying full text indexes, vector search etc?

Reply:
Thanks for the review. The concern here is that we arew also creating a structure that will be a standard into other pages as well. So having a descriptive on the operators, arguments, etc is useful. Maybe the syntax can be more flexible so if there are subcategories as it is for `createIndexes`, we can create it's own syntax.